### PR TITLE
Implement color type in input component

### PR DESCRIPTION
**Pull request by JavierCenteno**
Implementing type=color in lux-input

### CODE REVIEW

File: projects/lux/src/lib/helperFns.ts [hunk 1]
```diff
@@ -22,6 +22,21 @@ export const isValidUrl = (value: string): boolean => {
...
+  if (
+    value === 'currentcolor' ||
+    value === 'inherit' ||
+    value === 'initial' ||
+    value === 'revert' ||
+    value === 'unset'
+  ) {
+    return false;
+  }
+  return CSS.supports('color', value);
```
Comment (pjmolina) [inline projects/lux/src/lib/helperFns.ts:37]: This is smart!
https://developer.mozilla.org/en-US/docs/Web/API/CSS/supports
I didn't know it.

File: projects/lux/src/lib/helperFns.spec.ts [hunk 2]
```diff
@@ -83,6 +85,52 @@ describe('isValidEmail', () => {
...
+  });
+});
+
+describe('isValidColor', () => {
+  it('should return true for valid colors', () => {
+    expect(isValidColor('red')).toBeTrue();
+    expect(isValidColor('aqua')).toBeTrue();
+    expect(isValidColor('#000000')).toBeTrue();
+    expect(isValidColor('#FFFFFF')).toBeTrue();
+    expect(isValidColor('rgb(0,0,255)')).toBeTrue();
```
Comment (pjmolina) [inline projects/lux/src/lib/helperFns.spec.ts:111]: test also with spaces: example:  `'  rgb  ( 0 , 0,   255   )  '`

File: projects/lux/src/lib/helperFns.spec.ts [hunk 3]
```diff
@@ -83,6 +85,52 @@ describe('isValidEmail', () => {
...
+    expect(
+      isValidUrl('https://www.example.com/example/?example=true&ejemplo=false')
+    ).toBeTrue();
+    expect(isValidUrl('https://www.example.com/example123')).toBeTrue();
+  });
+  it('should return false for invalid URLs', () => {
+    expect(isValidUrl('a')).toBeFalse();
+    expect(isValidUrl('http//www.com')).toBeFalse();
+    expect(isValidUrl('http://www com')).toBeFalse();
+    expect(isValidUrl('https://www.com//')).toBeFalse();
```
Comment (pjmolina) [inline projects/lux/src/lib/helperFns.spec.ts]: this is valid (see test execution)

File: projects/lux/src/lib/helperFns.spec.ts [hunk 4]
```diff
@@ -83,6 +85,52 @@ describe('isValidEmail', () => {
...
+    expect(isValidColor('null')).toBeFalse();
+    expect(isValidColor('currentcolor')).toBeFalse();
+    expect(isValidColor('inherit')).toBeFalse();
+    expect(isValidColor('initial')).toBeFalse();
+    expect(isValidColor('revert')).toBeFalse();
+    expect(isValidColor('unset')).toBeFalse();
+    expect(isValidColor('Initial')).toBeFalse();
+    expect(isValidColor('UNSET')).toBeFalse();
+    expect(isValidColor('color')).toBeFalse();
+    expect(isValidColor('#0123')).toBeFalse();
```
Comment (pjmolina) [inline projects/lux/src/lib/helperFns.spec.ts]: this is valid: see test execution. valid format `#rgba`  with alpha channel

File: projects/lux/src/lib/helperFns.spec.ts [hunk 5]
```diff
@@ -83,6 +85,52 @@ describe('isValidEmail', () => {
...
+    expect(isValidColor('#GGGGGG')).toBeFalse();
+    expect(isValidColor('#01234567')).toBeFalse();
```
Comment (pjmolina) [inline projects/lux/src/lib/helperFns.spec.ts]: this is valid. see test execution: due to alpha channel:    `#rrggbbaa`

**Review by pjmolina — CHANGES_REQUESTED** [review]
some changes needed. broken tests

File: projects/lux/src/lib/helperFns.spec.ts (hunk 2 above)
Comment (JavierCenteno) [reply projects/lux/src/lib/helperFns.spec.ts:111]: Added: https://github.com/metadevpro/lux/pull/175/commits/39e52a6faa602556b59a579b4e341371e48ab627

**Pull request merged**